### ISSUE TITLE
Issue with Calamares crash on Virtualbox VM/Ryzen host.

### ISSUE THREAD

**Issue by bollio** [calamares/calamares#786]
[manjaro@manjaro ~]$ sudo -E calamares -d
23:58:13 [1]: Calamares version: 3.1.0 
Translation: Calamares: Using system locale: "en_US"
Translation: Qt: Using default locale, system locale one not found: "en_US"
23:58:13 [1]: Default font ===== 
Pixel size:    -1 
Point size:    10 
Point sizeF:   10 
Font family:   "Noto Sans" 
Metric height: 18 
23:58:13 [1]: Font height: 18 
23:58:13 [1]: Available languages: ("ar", "ast", "bg", "ca", "cs_CZ", "da", "de", "el", "en", "en_GB", "es_MX", "es", "eu", "fr", "hr", "hu", "id", "is", "it_IT", "ja", "lt", "nl", "pl", "pt_BR", "pt_PT", "ro", "ru", "sk", "sv", "th", "tr_TR", "zh_CN", "zh_TW") 
23:58:13 [1]: CalamaresApplication thread: 0x1d7f9d0 
23:58:13 [1]: Using log file: "/home/manjaro/.cache/Calamares/Calamares/Calamares.log" 
23:58:13 [1]: Using Calamares settings file at "/etc/calamares/settings.conf" 
23:58:13 [1]: Using Calamares branding file at "/usr/share/calamares/branding/manjaro/branding.desc" 
23:58:13 [1]: WARNING: the selected branding component does not ship translations. 
23:58:13 [1]: Loaded branding component "manjaro" 
23:58:13 [1]: STARTUP: initQmlPath, initSettings, initBranding done 
23:58:13 [1]: STARTUP: initModuleManager: module init started 
23:58:13 [1]: STARTUP: initModuleManager: all modules init done 
23:58:13 [1]: STARTUP: initJobQueue done 
23:58:13 [1]: Proposed window size: 1080 648 
23:58:13 [1]: STARTUP: CalamaresWindow created; loadModules started 
23:58:13 [1]: Initial locale  "en_US" 
23:58:13 [8]: Translation: Calamares: Using system locale: "en_US"
23:58:13 [8]: Translation: Qt: Using default locale, system locale one not found: "en_US"
23:58:13 [1]: ViewModule "welcome@welcome" loading complete.                                                                        
23:58:14 [1]: WARNING: cannot acquire a list of available locales. The locale and localecfg modules will be broken as long as this system does not provide  * a /usr/share/i18n/SUPPORTED file      OR  * a well-formed /etc/locale.gen     OR  * a complete pre-compiled locale-gen database which allows complete locale -a output.                                                                       
23:58:14 [1]: ViewModule "locale@locale" loading complete.                                                                          
23:58:14 [1]: ViewModule "keyboard@keyboard" loading complete.                                                                      
23:58:14 [8]: Loaded backend plugin:  "pmlibpartedbackendplugin"                                                                    
23:58:14 [1]: ViewModule "partition@partition" loading complete.                                                                    
23:58:14 [1]: ViewModule "users@users" loading complete.                                                                            
23:58:14 [1]: ViewModule "summary@summary" loading complete.                                                                        
23:58:14 [1]: Module "partition@partition" already loaded.                                                                          
23:58:14 [1]: QML import paths: ("/usr/share/calamares/qml", "/usr/bin", "qrc:/qt-project.org/imports", "/usr/lib/qt/qml")          
23:58:14 [1]: LAST VS IS EVS!                                                                                                       
23:58:14 [1]: LAST VS IS EVS!                                                                                                       
23:58:14 [1]: LAST VS IS EVS!                                                                                                       
23:58:14 [1]: LAST VS IS EVS!                                                                                                       
23:58:14 [1]: LAST VS IS EVS!                                                                                                       
23:58:14 [1]: Module "locale@locale" already loaded.                                                                                
23:58:14 [1]: LAST VS IS EVS!                                                                                                       
23:58:14 [1]: Module "keyboard@keyboard" already loaded.                                                                            
23:58:14 [1]: LAST VS IS EVS!                                                                                                       
23:58:14 [1]: LAST VS IS EVS!                                                                                                       
23:58:14 [1]: LAST VS IS EVS!                                                                                                       
23:58:14 [1]: LAST VS IS EVS!                                                                                                       
23:58:14 [1]: LAST VS IS EVS!                                                                                                       
23:58:14 [1]: LAST VS IS EVS!                                                                                                       
23:58:14 [1]: Module "users@users" already loaded.                                                                                  
23:58:14 [1]: LAST VS IS EVS!                                                                                                       
23:58:14 [1]: LAST VS IS EVS!                                                                                                       
23:58:14 [1]: LAST VS IS EVS!                                                                                                       
23:58:14 [1]: LAST VS IS EVS!                                                                                                       
23:58:14 [1]: LAST VS IS EVS!                                                                                                       
23:58:14 [1]: LAST VS IS EVS!                                                                                                       
23:58:14 [1]: LAST VS IS EVS!                                                                                                       
23:58:14 [1]: LAST VS IS EVS!                                                                                                       
23:58:14 [0]: void Calamares::Module::loadConfigurationFile(const QString&) bad module configuration format "/usr/share/calamares/modules/umount.conf"                                                                                                                  
23:58:14 [1]: LAST VS IS EVS!                                                                                                       
23:58:14 [1]: ViewModule "finished@finished" loading complete.                                                                      
23:58:14 [1]: STARTUP: loadModules for all modules done                                                                             
23:58:14 [1]: STARTUP: Window now visible and ProgressTreeView populated                                                            
23:58:14 [1]: Need at least storage bytes: 8482560409                                                                               
23:58:15 [8]: getting smart status failed for  "/dev/sda" :  Operation not supported                                                
23:58:15 [1]: Need at least ram bytes: 1073741824                                                                                   
23:58:15 [1]: enoughStorage, enoughRam, hasPower, hasInternet, isRoot:  true true true true true                                    
23:58:15 [1]: LIST OF DETECTED DEVICES:                                                                                             
node    capacity        name    prettyName                                                                                          
23:58:15 [1]: "/dev/sda" 107372805120 "ATA VBOX HARDDISK" "ATA VBOX HARDDISK – 100.00 GiB (/dev/sda)"                               
No layout variant specified on the command line                                                                                     
Trailing -variant option ignored                                                                                                    
23:58:15 [1]: xkbmap selection changed to:  "us" - ""                                                                               
23:58:15 [1]: "Osprober lines, clean:\n"                                                                                            
23:58:15 [1]: Updating partitioning state widgets.                                                                                  
23:58:15 [0]: QCoreApplication::postEvent: Unexpected null receiver                                                                 
23:58:15 [0]: QObject::setParent: Cannot set parent, new parent is in a different thread                                            
23:58:15 [1]: Required  storage B: 9019431321 "(8GB)"                                                                               
23:58:15 [1]: Storage capacity  B: 107374148096 "(99GB)" for ""    length: 209715133                                                
23:58:15 [1]: Partition "" authorized for replace install.                                                                          
23:58:15 [1]: Updating partitioning preview widgets.                                                                                
23:58:15 [0]: QCoreApplication::postEvent: Unexpected null receiver                                                                 
terminate called after throwing an instance of 'YAML::ParserException'                                                              
  what():  yaml-cpp: error at line 21, column 80: illegal map value                                                                 
Aborted                                                                                                                             
[manjaro@manjaro ~]

**Comment by bollio** [comment]
I've tried both thin and static virtual HDD.  EFI/bios and played with various other settings.  Just seems to crash as soon as the window pops up every time.

**Comment by philmmanjaro** [comment]
`Calamares version: 3.1.0` is an older version. I assume you also have not the latest ISO of Manjaro. Please check out [v17.0.3](https://manjaro.org/get-manjaro/) which ships **v3.1.3**. Also I run Ryzen on my own without problems.

**Comment by adriaandegroot** [comment]
The message at the end of the log,
```
terminate called after throwing an instance of 'YAML::ParserException'
what(): yaml-cpp: error at line 21, column 80: illegal map value
```
says that your configuration file is not valid; this is the same message that was in #695, which was eventually closed after a long and confusing discussion. In the meantime there's newer Manjaro ISO's and newer Calamares releases, which fix up a number of early crashes in special cases.

The message when the YAML is wrong could be improved, though.

Closing as dup of #686, #695.

**Comment by bollio** [comment]
[manjaro@manjaro ~]$ sudo -E calamares -d
QStandardPaths: wrong ownership on runtime directory /run/user/1000, 1000 instead of 0
08:49:49 [1]: Calamares version: 3.1.3 
Translation: Calamares: Using system locale: "en_US"
08:49:49 [1]: Default font ===== 
Pixel size:    -1 
Point size:    10 
Point sizeF:   10 
Font family:   "Noto Sans" 
Metric height: 18 
08:49:49 [1]: Font height: 18 
08:49:49 [1]: Available languages: ("ar", "ast", "bg", "ca", "cs_CZ", "da", "de", "el", "en", "en_GB", "es_MX", "es", "eu", "fr", "he", "hr", "hu", "id", "is", "it_IT", "ja", "lt", "nl", "pl", "pt_BR", "pt_PT", "ro", "ru", "sk", "sv", "th", "tr_TR", "zh_CN", "zh_TW") 
08:49:49 [1]: CalamaresApplication thread: 0x55b34384db20 
08:49:49 [1]: Using log file: "/home/manjaro/.cache/Calamares/Calamares/Calamares.log" 
08:49:49 [1]: Using Calamares settings file at "/etc/calamares/settings.conf" 
08:49:49 [1]: Using Calamares branding file at "/usr/share/calamares/branding/manjaro/branding.desc" 
08:49:49 [1]: WARNING: the selected branding component does not ship translations. 
08:49:49 [1]: Loaded branding component "manjaro" 
08:49:49 [1]: STARTUP: initQmlPath, initSettings, initBranding done 
08:49:49 [1]: STARTUP: initModuleManager: module init started 
08:49:49 [1]: STARTUP: initModuleManager: all modules init done 
08:49:49 [1]: STARTUP: initJobQueue done 
08:49:49 [1]: Available size QSize(1080, 1813) 
08:49:49 [1]:   Proposed window size: 1080 648 
08:49:49 [1]: STARTUP: CalamaresWindow created; loadModules started 
08:49:49 [1]: Initial locale  "en_US" 
08:49:49 [8]: Translation: Calamares: Using system locale: "en_US"
08:49:49 [1]: Welcome string false "Manjaro Linux 17.0 \"Gellivara\"" 
08:49:49 [1]: ViewModule "welcome@welcome" loading complete. 
08:49:49 [1]: WARNING: guessLocaleConfiguration can't guess from an empty list. 
08:49:49 [1]: ViewModule "locale@locale" loading complete. 
08:49:49 [1]: ViewModule "keyboard@keyboard" loading complete. 
08:49:49 [8]: Loaded backend plugin:  "pmlibpartedbackendplugin"
08:49:49 [1]: ViewModule "partition@partition" loading complete. 
08:49:49 [1]: ViewModule "users@users" loading complete. 
08:49:49 [1]: ViewModule "summary@summary" loading complete. 
08:49:49 [1]: Module "partition@partition" already loaded. 
08:49:49 [1]: QML import paths: ("/usr/share/calamares/qml", "/usr/bin", "qrc:/qt-project.org/imports", "/usr/lib/qt/qml") 
08:49:49 [1]: Module "locale@locale" already loaded. 
08:49:49 [1]: Module "keyboard@keyboard" already loaded. 
08:49:49 [1]: Module "users@users" already loaded. 
08:49:49 [1]: FinishedPage() 
08:49:49 [1]: FinishedViewStep() 
08:49:49 [1]: FinishedViewStep::widget() 
08:49:49 [1]: FinishedViewStep::widget() 
08:49:49 [1]: FinishedViewStep::widget() 
08:49:49 [1]: ViewModule "finished@finished" loading complete. 
08:49:49 [1]: STARTUP: loadModules for all modules done 
08:49:50 [1]: STARTUP: Window now visible and ProgressTreeView populated 
08:49:50 [1]: Need at least storage bytes: 8482560409 
08:49:50 [1]: Need at least ram bytes: 1073741824 
08:49:50 [1]: enoughStorage, enoughRam, hasPower, hasInternet, isRoot:  true true true true true 
08:49:50 [8]: getting smart status failed for  "/dev/sda" :  Operation not supported
08:49:50 [1]: xkbmap selection changed to:  "us" - "" 
08:49:50 [1]: Removing unsuitable devices: 1 candidates. 
08:49:50 [1]: LIST OF DETECTED DEVICES: 
08:49:50 [1]: node      capacity        name    prettyName 
08:49:50 [1]: "/dev/sda" 107372805120 "ATA VBOX HARDDISK" "ATA VBOX HARDDISK – 100.00 GiB (/dev/sda)" 
08:49:50 [1]: .. 1 devices detected. 
08:49:50 [1]: "Osprober lines, clean:\n" 
08:49:50 [1]: Updating partitioning state widgets. 
08:49:50 [0]: QCoreApplication::postEvent: Unexpected null receiver
08:49:50 [0]: QObject::setParent: Cannot set parent, new parent is in a different thread
08:49:50 [1]: Required  storage B: 9019431321 "(8GB)" 
08:49:50 [1]: Storage capacity  B: 107374148096 "(99GB)" for ""    length: 209715133 
08:49:50 [1]: Partition "" authorized for replace install. 
08:49:50 [1]: Updating partitioning preview widgets. 
08:49:50 [0]: QCoreApplication::postEvent: Unexpected null receiver
terminate called after throwing an instance of 'YAML::ParserException'
  what():  yaml-cpp: error at line 21, column 80: illegal map value
Aborted

**Comment by bollio** [comment]
same issue with different iso

**Comment by adriaandegroot** [comment]
By now I really want to know which yaml file is causing that exception :|

**Comment by adriaandegroot** [comment]
The workaround mentioned in the other issue is to disable networking during the install; I also see a bit of code in the NetInstall parts that *doesn't* handle YAML errors at all, so maybe you're getting a broken response from Manjaro's servers, and that's crashing Calamares. I'll work on better logging / error handling there, and can you try the no-networking workaround?

**Comment by bollio** [comment]
Sure enough the work around was flawless.  Without internet access, Calamares stayed open and installed flawlessly.  

My system clock and timezone was accurate in the live CD this time right as well... on previous attempts, it was inaccurate.  Not sure if that helps, but something I noticed.

**Comment by adriaandegroot** [comment]
@philmmanjaro Looks like the Manjaro servers are, at least sometimes, returning .. ooh, could be the GeoIP code, too, which doesn't handle YAML parse errors, either.

**Comment by philmmanjaro** [comment]
@bollio: which exact ISO was used? Some of them have **netinstall** activated. Our official releases **XFCE**, **KDE** and **Gnome** not. When does the crash happen?

**Comment by bollio** [comment]
@philmmanjaro I encountered the issues with Manjaro KDE 17.0.1 and 17.0.3.   Checking to see if I can reproduce in GNOME and XFCE this morning, on first run on Calamares it exhibits the exact same behavior as I had encountered with KDE (being an immediate close).  However, now if I run Calamares again after the first crash, Calamares will freeze and then ultimately crash 30+ seconds later.  This behavior is the shared between XFCE and GNOME 17.0.3.
